Libraries/LibGfx/Bitmap.cpp

```diff
@@ -44,11 +44,16 @@
 
 namespace Gfx {
 
-static bool size_would_overflow(BitmapFormat format, const IntSize& size)
+static bool size_would_overflow(BitmapFormat, const IntSize& size)
 {
     if (size.width() < 0 || size.height() < 0)
         return true;
-    return Checked<size_t>::multiplication_would_overflow(size.width(), size.height(), Bitmap::bpp_for_format(format));
+    // This check is a bit arbitrary, but should protect us from most shenanigans:
+    if (size.width() >= 16384 || size.height() >= 16384)
+        return true;
+    // This check is absolutely necessary. Note that Bitmap::Bitmap always stores
+    // data as RGBA32 internally, so currently we ignore the indicated format.
```

Comment:
This is not accurate. `Bitmap` respects the `BitmapFormat`. Don't be fooled by the `RGBA32*`, it's cast to `u8*` for indexed 8-bit bitmaps.

Reply:
Interesting! We do seem to currently allocate that amount of memory, *even if* we only use it for `u8`s. Is that a bug?

// In Bitmap.h:
```
    size_t size_in_bytes() const { return m_pitch * m_size.height(); }
```

// In Bitmap.cpp:
```
Bitmap::Bitmap(BitmapFormat format, const IntSize& size, Purgeable purgeable)
…
    , m_pitch(round_up_to_power_of_two(size.width() * sizeof(RGBA32), 16))
…
{
…
    m_data = (RGBA32*)mmap_with_name(nullptr, size_in_bytes(), …);
…
}
```

Reply:
That's a bug :)

Reply:
Fixed it :)